docs/ci.md

```diff
@@ -0,0 +1,44 @@
+# CI
+
+## Parallelization with Parallel Tests & Knapsack
+Knapsack and Parallel Tests gems allow us to run tests in several nodes at the same time, benefiting us in the execution time. Knapsack parallelizes them at node level while Parallel Tests does it at CPU level.
+
+Knapsack splits tests based on an execution time report. In case there are files that were not added in the report, they will all run on the same node and may overload it, so it is strongly recommended to update the report frequently.
+
+## Configuration
+In case you want to use this you will need a script that split spec files called `parallel_tests` which sets up the configuration, assuming you have `n_nodes * cpu_cores_quantity`.
+
+On Github Actions you can add any nodes you want using matrix strategy, setting up some variables:
+
+```sh
+  ci_node_total: [4]
+  # set N-1 indexes for parallel jobs
+  # When you run 2 parallel jobs then first job will have index 0, the second job will have index 1 etc
+  ci_node_index: [0, 1, 2, 3]
+```
+
+CPU cores quantity on every node are obtained automatically from Github Actions config `echo "cpu_cores=$(nproc)" >> $GITHUB_ENV`.
+
+If you want to update it manually you can do it by updating this variable:
+`PARALLEL_TESTS_CONCURRENCY: 2`
+
+To update tests on local machine you can by executing `KNAPSACK_CI_NODE_TOTAL=4 KNAPSACK_CI_NODE_INDEX=1 PARALLEL_TESTS_CONCURRENCY=2 bundle exec parallel_rspec -n 2 -e './bin/parallel_tests'`. This will run subset of tests files corresponding to second node.
```

Comment:
```suggestion
To update tests on local machine you can execute `KNAPSACK_CI_NODE_TOTAL=4 KNAPSACK_CI_NODE_INDEX=1 PARALLEL_TESTS_CONCURRENCY=2 bundle exec parallel_rspec -n 2 -e './bin/parallel_tests'`. This will run subset of tests files corresponding to second node.
```